src/google/google-calendars/googlecalendarsyncadaptor.cpp

```diff
@@ -1241,327 +1305,351 @@ void GoogleCalendarSyncAdaptor::finishedRequestingRemoteEvents(int accountId, co
+                for (int i = 0; i < addedList.size(); ++i) {
+                    const QString &gcalId(gCalEventId(addedList[i]));
+                    if (gcalId == eventId) {
+                        SOCIALD_LOG_DEBUG("Discarding local event addition:" << addedList[i]->uid() << "due to remote EXDATE addition.  Sub-optimal resolution strategy!");
+                        addedList.remove(i);
+                        break;
+                    }
+                }
+            } else {
+                // remote deleted event was also deleted locally, can ignore.
+                SOCIALD_LOG_DEBUG("Event deleted remotely:" << eventId << "was already deleted locally; discarding unneeded local deletion from upsync");
+                discardedLocalRemovals++;
+            }
         } else if (deletedMap.contains(eventId)) {
-            // event was deleted locally, can ignore.
-            SOCIALD_LOG_DEBUG("Event deleted remotely:" << eventId << "was already deleted locally; ignoring");
+            // remote change will be discarded due to local deletion.
+            SOCIALD_LOG_DEBUG("Discarding remote event modification:" << eventId << "due to local deletion");
+            discardedRemoteModifications++;
         } else if (allMap.contains(eventId)) {
-            // modify existing event.
-            SOCIALD_LOG_DEBUG("Event modified remotely:" << eventId);
+            // remote modification of an existing event.
             KCalCore::Event::Ptr event = allMap.value(eventId);
-            // then, update local event appropriately.
-            bool changed = false;
-            jsonToKCal(eventData, event, m_serverCalendarIdToDefaultReminderTimes[accountId].value(calendarId), m_icalFormat, &changed);
-            if (changed) {
-                remoteModified++;
-                m_storageNeedsSave = true;
-                // if both local and server were modified, prefer server.
-                updatedMap.remove(eventId);
+            bool changed = remoteModificationIsReal(eventData, event);
+            if (partialUpsyncArtifactsNeedingUpdate.contains(eventId)) {
+                // This event was partially upsynced and then connectivity died before we committed
+                // and updated its comment field with the gcalId it was given by the remote server.
+                // During this sync cycle, we will update it by assuming remote modification.
+                // Note: this will lose any local changes made since it was partially-upsynced,
+                // however the alternative is to lose remote changes made since then...
+                // So we stick with our "prefer-remote" conflict resolution strategy here.
+                SOCIALD_LOG_DEBUG("Reloading partial upsync artifact:" << eventId << "from server as a modification");
+                changed = true; // actually a no-op, the etag/gcalId comment field change will ensure this is set above.
```

Comment:
If it must be true in this case and the debug is required, how about 

 if (partialUpsyncArtifactsNeedingUpdate.contains(eventId)) {
    // debug
     changed = true;
} else {
     changed = remoteModificationIsReal(eventData, event);
}

If the debug is not required then
bool changed = partialUpsyncArtifactsNeedingUpdate.contains(eventId) || remoteModificationIsReal(eventData, event)